src/logic/safe/store/actions/transactions/fetchTransactions/loadGatewayTransactions.ts

```diff
@@ -0,0 +1,93 @@
+import axios, { AxiosResponse } from 'axios'
+
+import { getSafeClientGatewayBaseUrl } from 'src/config'
+import { HistoryGatewayResponse, QueuedGatewayResponse } from 'src/logic/safe/store/models/types/gateway'
+import { checksumAddress } from 'src/utils/checksumAddress'
+
+/*************/
+/*  HISTORY  */
+/*************/
+const getHistoryTransactionsUrl = (safeAddress: string): string => {
+  const address = checksumAddress(safeAddress)
+  return `${getSafeClientGatewayBaseUrl(address)}/transactions/history/`
+}
+
+const historyPointers: { [safeAddress: string]: { next: string | null; previous: string | null } } = {}
+
+export const loadPagedHistoryTransactions = async (
+  safeAddress: string,
+): Promise<{ values: HistoryGatewayResponse['results']; next: string | null } | undefined> => {
+  // if `historyPointers[safeAddress] is `undefined` it means `loadHistoryTransactions` wasn't called
+  // if `historyPointers[safeAddress].next is `null`, it means it reached the last page in gateway-client
+  if (!historyPointers[safeAddress]?.next) {
+    return
+  }
+
+  const {
+    data: { results, ...pointers },
+  } = await axios.get<HistoryGatewayResponse, AxiosResponse<HistoryGatewayResponse>>(
+    historyPointers[safeAddress].next ?? '',
+  )
+
+  historyPointers[safeAddress] = pointers
+
+  return { values: results, next: historyPointers[safeAddress].next }
+}
+
+export const loadHistoryTransactions = async (safeAddress: string): Promise<HistoryGatewayResponse['results']> => {
+  const historyTransactionsUrl = getHistoryTransactionsUrl(safeAddress)
+
+  const {
+    data: { results, ...pointers },
+  } = await axios.get<HistoryGatewayResponse, AxiosResponse<HistoryGatewayResponse>>(historyTransactionsUrl)
+
+  if (!historyPointers[safeAddress]) {
+    historyPointers[safeAddress] = pointers
+  }
+
+  return results
+}
+
+/************/
+/*  QUEUED  */
+/************/
+const getQueuedTransactionsUrl = (safeAddress: string): string => {
+  const address = checksumAddress(safeAddress)
+  return `${getSafeClientGatewayBaseUrl(address)}/transactions/queued/`
+}
+
+const queuedPointers: { [safeAddress: string]: { next: string | null; previous: string | null } } = {}
+
+export const loadPagedQueuedTransactions = async (
+  safeAddress: string,
+): Promise<{ values: QueuedGatewayResponse['results']; next: string | null } | undefined> => {
+  // if `historyPointers[safeAddress] is `undefined` it means `loadHistoryTransactions` wasn't called
+  // if `historyPointers[safeAddress].next is `null`, it means it reached the last page in gateway-client
+  if (!queuedPointers[safeAddress]?.next) {
+    return
+  }
+
+  const {
+    data: { results, ...pointers },
+  } = await axios.get<QueuedGatewayResponse, AxiosResponse<QueuedGatewayResponse>>(
+    queuedPointers[safeAddress].next ?? '',
```

Comment:
This is confusing, in the above validation we are checking the next value, so if the code reaches this line it should not be falsy. I suggest to it to this: `queuedPointers[safeAddress].next as string`.